primaryHandler.go

```diff
@@ -322,7 +326,7 @@ func NewPrimaryHandler(logger log.Logger, v *viper.Viper, registry xmetrics.Regi
 					}
 				}
 				return kv
-			},
+			}, candlelight.InjectTraceInformationInLogger(traceConfig.HeaderConfig),
```

Comment:
I'm confused - isn't the tracing information already in the logger with changes made to the `SetLogger()` function?

Reply:
In `SetLogger()` we are adding the trace information which will be used when the request is made.
but at line 314 we are not using `GetLogger` that's why I need to add it here as well.
https://github.com/xmidt-org/scytale/blob/bce89b9d1373dfb0a30ac47c0faa78494abb3c8e/primaryHandler.go#L313-L329

Reply:
I investigated this further, and it's part of a pre-existing bug in our middleware chaining:
https://github.com/xmidt-org/scytale/issues/130

No need to make any changes in your pr - what you did makes sense for the current logic. 👍